tests/test_lgbm_compression.py

```diff
@@ -20,7 +24,36 @@ def lgbm_regressor(rng):
     return LGBMRegressor(random_state=rng)
 
 
-def test_compresed_predictions(diabetes_toy_df, lgbm_regressor, tmp_path):
+@pytest.fixture
+def lgbm_regressor_linear(rng):
+    return LGBMRegressor(random_state=rng, linear_trees=True)
+
+
+@pytest.mark.xfail(reason="reconstructed model string is not expected to be equal")
+def test_model_string_equality(diabetes_toy_df, lgbm_regressor):
+    """
+    This test should fail because the model string
+    will not be equivalent since we're omitting values.
+    Nevertheless, for required features, some string rows should be equal.
+    This helps in debugging that.
```

Comment:
Why not make this a normal test that checks the rows that should be equal for equality and the others for inequality? Is it because it's too much effort for little benefit?

Reply:
We'd (again) have to parse out all the matching rows (for every tree) and then check
I think it's too much work for what it is used for (a debugging helper)

Should we let it in, though? I'd otherwise remove it agian

Reply:
Let's leave it out for now.